auction/msgp_gen.go

```diff
@@ -0,0 +1,3405 @@
+package auction
+
+// Code generated by github.com/algorand/msgp DO NOT EDIT.
+
+import (
+	"github.com/algorand/msgp/msgp"
+)
+
+// MarshalMsg implements msgp.Marshaler
+func (z *Bid) MarshalMsg(b []byte) (o []byte, err error) {
```

Comment:
Out of curiosity, how do we determine whether the type has a pointer receiver or a value one?

Reply:
For mutable methods (i.e., `UnmarshalMsg`), it's always pointer receiver.  For immutable methods, there are some heuristics for deciding what to do:

https://github.com/algorand/msgp/blob/master/gen/spec.go#L272